addons/website_event_booth_sale/tests/test_event_booth_sale.py

```diff
@@ -85,3 +91,39 @@ def test_booth_pricelists_different_currencies(self):
             'phone': '+1 555-555-5555',
         })
         self.start_tour("/web", 'event_booth_sale_pricelists_different_currencies', login='admin')
+
+    def test_zero_price_product_rule(self):
+        """
+        With the `prevent_zero_price_sale` that we have on website, we can't add free products
+        to our cart.
+        There is an exception for certain product types specified by the
+        `_get_product_types_allow_zero_price` method, so this test ensures that it works
+        by mocking that function to return the "service" product type.
+        """
+        WebsiteSaleController = WebsiteSale()
+        website_prevent_zero_price = self.env['website'].create({
+            'name': 'Prevent zero price sale',
+            'prevent_zero_price_sale': True,
+        })
+        product_consu = self.env['product.product'].create({
+            'name': 'Cannot be zero price',
+            'type': 'consu',
+            'list_price': 0,
+            'website_published': True,
+        })
+        product_service = self.env['product.product'].create({
+            'name': 'Can be zero price',
+            'type': 'service',
+            'service_tracking': 'event_booth',
+            'list_price': 0,
+            'website_published': True,
+        })
+
+        with patch.object(ProductTemplate, '_get_product_types_allow_zero_price', lambda pt: ['event_booth']):
```

Comment:
If you move the test here, there is no need for this patch anymore.
You should keep it `website_sale`, make it so that the patch return 'no', so that any zero priced product can be added, and check that without the patch, they cannot.